_posts/2021-06-28-hhvm-4.116.markdown

```diff
@@ -0,0 +1,50 @@
+---
+title: "HHVM 4.116"
+layout: post
+author: jjergus
+category: blog
+---
+
+HHVM 4.116 is released! This release marks the end of support for 4.110;
+HHVM 4.111&ndash;4.115 remain supported, as do the 4.80 and 4.102 LTS releases.
+
+# Highlights
+
+- [`HH_FIXME\UNSAFE_CAST`](https://docs.hhvm.com/hack/silencing-errors) can now
```

Comment:
I'll update this to link to the correct section after https://github.com/hhvm/user-documentation/pull/1046 is merged.

Reply:
An important note for library authors. If your support window includes both 4.116 and 4.117-dev, you can not use unsafe cast yet. Unsafe cast will likely be renamed in hhvm 4.117. https://github.com/facebook/hhvm/commit/07acc88f4bf2aaf71f97e12f138b603824d4df11

There is also an hhconfig option to disallow the use of unsafe cast. We strongly recommend not touching this option. Open source hack libraries may soon start to include unsafe casts, which will leave you unable to upgrade. https://github.com/facebook/hhvm/blob/master/hphp/hack/src/options/globalOptions.ml#L861

Reply:
Also a behavior which is likely a bug. The second argument of unsafe cast MUST be a literal string (good so far) which can not include an odd amount of percent signs in succession.

I hope to be able to fix this issue by adding an interface to hhvm which declares all the magic string printf sequences to return nothing and to change the second argument of unsafe cast to take a format string of that type, instead of a format string of `nothing`.

Reply:
Good catch, I'll postpone the announcement until next week so we can announce it with the final name.

I'm not sure if adding the interface is worth the extra complexity just to be able to use % in the string. Like HH_FIXMEs, this is not meant to be used in new code so I don't think we care much about making it perfect...